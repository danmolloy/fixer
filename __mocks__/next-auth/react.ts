import { mockSession } from '../session';

module.exports = {
  __esModule: true,
  ...jest.requireActual('next-auth/react'),
  auth: jest.fn(() => {
    return mockSession;
  }),
  signIn: jest.fn(),
  signOut: jest.fn(),
  useSession: jest.fn(),
};
