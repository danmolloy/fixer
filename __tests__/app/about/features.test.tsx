import { act, render, screen } from '@testing-library/react';
import AboutFeatures, { fixerFeatureList } from '../../../app/about/features';
import '@testing-library/jest-dom';

describe('<AboutFeatures />', () => {
  beforeEach(() => {
    render(<AboutFeatures />);
  });
  it('<AboutFeatures /> renders', () => {
    const aboutFeatures = screen.getByTestId('about-features');
    expect(aboutFeatures).toBeInTheDocument();
  });
  it('fixerFeatures list is in the document', () => {
    const aboutFeatures = screen.getByTestId('about-features');

    for (let i = 0; i < fixerFeatureList.length; i++) {
      expect(aboutFeatures.textContent).toMatch(fixerFeatureList[i].title);
      expect(aboutFeatures.textContent).toMatch(fixerFeatureList[i].blurb);
    }
  });
  it('matches snapshot', () => {
    const aboutFeatures = screen.getByTestId('about-features');
    expect(aboutFeatures).toMatchSnapshot();
  });
});
