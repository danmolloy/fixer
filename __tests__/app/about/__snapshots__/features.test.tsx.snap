// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AboutFeatures /> matches snapshot 1`] = `
<div
  class=""
  data-testid="about-features"
>
  <div
    class="pb-6"
  >
    <p
      class="text-indigo-600 text-xl"
    >
      Fix faster
    </p>
    <h2
      class="text-3xl  py-2"
    >
      Everything you need to book an orchestra
    </h2>
  </div>
  <p
    class="py-8"
  >
    Aliquet nec orci mattis amet quisque ullamcorper neque, nibh sem. At arcu, sit dui mi, nibh dui, diam eget aliquam. Quisque id at vitae feugiat egestas ac. Diam nulla orci at in viverra scelerisque eget. Eleifend egestas fringilla sapien.
  </p>
  <div
    class="border-l border-l-indigo-600 px-8"
  >
    <p
      class="font-bold pb-4"
    >
      “Vel ultricies morbi odio facilisi ultrices accumsan donec lacus purus. Lectus nibh ullamcorper ac dictum justo in euismod. Risus aenean ut elit massa. In amet aliquet eget cras. Sem volutpat enim tristique.”
    </p>
    <div
      class="flex flex-row items-center"
    >
      <div
        class="ml-4"
      >
        <p
          class="font-bold"
        >
          Roy Dereks
        </p>
        <p
          class="text-gray-400"
        >
          Fixer
        </p>
      </div>
    </div>
  </div>
  <p
    class="py-8"
  >
    Faucibus commodo massa rhoncus, volutpat. Dignissim sed eget risus enim. Mattis mauris semper sed amet vitae sed turpis id. Id dolor praesent donec est. Odio penatibus risus viverra tellus varius sit neque erat velit. Faucibus commodo massa rhoncus, volutpat. Dignissim sed eget risus enim. Mattis mauris semper sed amet vitae sed turpis id.
  </p>
  <div
    class="flex flex-col px-6 md:flex-row flex-wrap"
  >
    <div
      class="flex flex-row  my-2 sm:my-4  md:w-1/2"
    >
      <div
        class="text-blue-600 text-2xl py-2 md:px-2"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 640 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M32,224H64V416H32A31.96166,31.96166,0,0,1,0,384V256A31.96166,31.96166,0,0,1,32,224Zm512-48V448a64.06328,64.06328,0,0,1-64,64H160a64.06328,64.06328,0,0,1-64-64V176a79.974,79.974,0,0,1,80-80H288V32a32,32,0,0,1,64,0V96H464A79.974,79.974,0,0,1,544,176ZM264,256a40,40,0,1,0-40,40A39.997,39.997,0,0,0,264,256Zm-8,128H192v32h64Zm96,0H288v32h64ZM456,256a40,40,0,1,0-40,40A39.997,39.997,0,0,0,456,256Zm-8,128H384v32h64ZM640,256V384a31.96166,31.96166,0,0,1-32,32H576V224h32A31.96166,31.96166,0,0,1,640,256Z"
          />
        </svg>
      </div>
      <div
        class="flex flex-col px-4"
      >
        <h3
          class="text-lg py-2"
        >
          Automated fixing
        </h3>
        <p
          class="text-slate-600 md:w-2/3"
        >
          Morbi viverra dui mi arcu sed. Tellus semper adipiscing suspendisse semper morbi. Odio urna massa nunc massa.
        </p>
      </div>
    </div>
    <div
      class="flex flex-row  my-2 sm:my-4  md:w-1/2"
    >
      <div
        class="text-blue-600 text-2xl py-2 md:px-2"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M21 5c0-1.103-.897-2-2-2H5c-1.103 0-2 .897-2 2v14c0 1.103.897 2 2 2h14c1.103 0 2-.897 2-2V5zM5 19V5h14l.002 14H5z"
          />
          <path
            d="M7 7h1.998v2H7zm4 0h6v2h-6zm-4 4h1.998v2H7zm4 0h6v2h-6zm-4 4h1.998v2H7zm4 0h6v2h-6z"
          />
        </svg>
      </div>
      <div
        class="flex flex-col px-4"
      >
        <h3
          class="text-lg py-2"
        >
          Gig interface
        </h3>
        <p
          class="text-slate-600 md:w-2/3"
        >
          Sit quis amet rutrum tellus ullamcorper ultricies libero dolor eget. Sem sodales gravida quam turpis enim lacus amet.
        </p>
      </div>
    </div>
    <div
      class="flex flex-row  my-2 sm:my-4  md:w-1/2"
    >
      <div
        class="text-blue-600 text-2xl py-2 md:px-2"
      >
        <svg
          fill="currentColor"
          height="1em"
          role="img"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title />
          <path
            d="M12 0A12 12 0 0 0 0 12a12 12 0 0 0 12 12 12 12 0 0 0 12-12A12 12 0 0 0 12 0zM4.911 7.089h11.456a2.197 2.197 0 0 1 2.165 2.19v5.863a2.213 2.213 0 0 1-2.177 2.178H8.04c-1.174 0-2.04-.99-2.04-2.178v-4.639L4.503 7.905c-.31-.42-.05-.816.408-.816zm3.415 2.19c-.347 0-.68.21-.68.544 0 .334.333.544.68.544h7.905c.346 0 .68-.21.68-.544 0-.334-.334-.545-.68-.545zm0 2.177c-.347 0-.68.21-.68.544 0 .334.333.544.68.544h7.905c.346 0 .68-.21.68-.544 0-.334-.334-.544-.68-.544zm-.013 2.19c-.346 0-.68.21-.68.544 0 .334.334.544.68.544h5.728c.347 0 .68-.21.68-.544 0-.334-.333-.545-.68-.545z"
          />
        </svg>
      </div>
      <div
        class="flex flex-col px-4"
      >
        <h3
          class="text-lg py-2"
        >
          Update the orchestra's diary in seconds
        </h3>
        <p
          class="text-slate-600 md:w-2/3"
        >
          Quisque est vel vulputate cursus. Risus proin diam nunc commodo. Lobortis auctor congue commodo diam neque.
        </p>
      </div>
    </div>
    <div
      class="flex flex-row  my-2 sm:my-4  md:w-1/2"
    >
      <div
        class="text-blue-600 text-2xl py-2 md:px-2"
      >
        <svg
          fill="currentColor"
          height="1em"
          role="img"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title />
          <path
            d="M12 0C5.375 0 0 5.375 0 12c0 6.627 5.375 12 12 12 6.626 0 12-5.373 12-12 0-6.625-5.373-12-12-12zm-.624 21.62v-7.528H7.19L13.203 2.38v7.528h4.029L11.376 21.62z"
          />
        </svg>
      </div>
      <div
        class="flex flex-col px-4"
      >
        <h3
          class="text-lg py-2"
        >
          Find new players in the directory
        </h3>
        <p
          class="text-slate-600 md:w-2/3"
        >
          Arcu egestas dolor vel iaculis in ipsum mauris. Tincidunt mattis aliquet hac quis. Id hac maecenas ac donec pharetra eget.
        </p>
      </div>
    </div>
  </div>
  <p
    class="py-8"
  >
    Et vitae blandit facilisi magna lacus commodo. Vitae sapien duis odio id et. Id blandit molestie auctor fermentum dignissim. Lacus diam tincidunt ac cursus in vel. Mauris varius vulputate et ultrices hac adipiscing egestas. Iaculis convallis ac tempor et ut. Ac lorem vel integer orci.
  </p>
</div>
`;
